store/postgres/src/relational/ddl.rs

```diff
@@ -198,6 +198,56 @@ impl Table {
         }
     }
 
+    /// Calculates the indexing method and expression for a database column.
+    ///
+    /// ### Parameters
+    /// * `immutable`: A boolean flag indicating whether the table is immutable.
+    /// * `column`: A reference to the `Column` struct, representing the database column for which the index method and expression are being calculated.
+    ///
+    /// ### Returns
+    /// A tuple `(String, String)` where:
+    /// - The first element is the indexing method ("btree", "gist", or "gin"),
+    /// - The second element is the index expression as a string.
+    pub fn calculate_index_method_and_expression(
+        immutable: bool,
+        column: &Column,
+    ) -> (String, String) {
```

Comment:
There's an enum for index methods [here](https://github.com/graphprotocol/graph-node/blob/master/store/postgres/src/relational/index.rs#L17) - it would be better to use that. I wonder if that should return the whole `CreateIndex` from that file instead of the second string, but that might become a bit too unwieldy. Maybe some or all of this logic should go into a new constructor for `CreateIndex`

Reply:
Actually, I think what I said above would be good for another refactor, but what you have here is ok for now.